dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+  "command": {
+    "_ObjectType": "CaptureCommand",      
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98"
+  }
+}
+
+```
+#### CaptureCommand Response
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a"    
+}
+```
+
+### VoidCommand
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|VOID_PAYMENT|Receiver > Sender|This functionality allows the receiver to void the payment. For example, the merchant voided the payment because the purchased products are out of stock. Relevant only for Auth / Capture payments|VoidCommand|Reference ID ; Receiver Status|Sender Status|
+|
+
+#### VoidCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|status|StatusObject|Y|Receiver status. For the void command the status should be abort and the abort code should be void|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "VoidCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "VoidCommand",
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "status": {
+      "status": "abort",
+      "abort_code": "void",
+      "abort_message": "abort message"
+    }
+  }
+}
+
+
+
+
+```
+#### VoidCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|StatusObject|Y|Sender status. Used to verify that the sender VASP acknoledges the void command by responding with the abort status|
+|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "StatusObject",
+    "status": {
+      "status": "abort"
+    }
+  }
+}
+```
+
+### AbortCommand
+
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|ABORT_PAYMENT|Sender > Receiver / Receiver > Sender|This functionality allows both parties to abort the payment. For example, the receiver may want to abort the payment following failed risk checks|AbortCommand|Reference ID ; Requester (Sender / Receiver) Status|Responder (Sender / Receiver) Status|
+|
+
+#### AbortCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|sender / receiver|PaymentActorObject|Y|Indicates the initiator of the abort command (can be either the sender or the receiver). Must contain a nested StatusObject (other elements are not required)|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "AbortCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "AbortCommand",    
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "sender": {
+      "status": {
+        "status":"abort",
```

Comment:
would also argue that having to pass around abort and embedding this inside a sender feels redundant.

Reply:
Same as the previous comment. Will update the payload structure